frontend/dockerfile/parser/errors.go

```diff
@@ -48,6 +65,56 @@ func WithLocation(err error, location []Range) error {
 	})
 }
 
+func MergeLocations(locations ...[]Range) []Range {
+	// normalize the range by looping through all the ranges
+	// and ensuring that the start and end are in the correct order
+	allRanges := []Range{}
+	for _, location := range locations {
+		for i, r := range location {
+			if r.Start.Line > r.End.Line {
+				location[i].Start, location[i].End = r.End, r.Start
+			}
+			allRanges = append(allRanges, location[i])
+		}
+	}
+	if len(allRanges) == 0 {
+		return []Range{}
+	}
+	if len(allRanges) == 1 {
+		return allRanges
+	}
+
+	// sort the locations by start line
+	// this is helpful for merging the ranges
+	sort.Slice(allRanges, func(i, j int) bool {
+		return allRanges[i].Start.Line < allRanges[j].Start.Line
+	})
+
+	mergedRanges := []Range{}
+	current := allRanges[0]
+	for i, r := range allRanges[1 : len(allRanges)-1] {
+		if r.Start.Line <= current.End.Line {
+			current.End.Line = int(math.Max(float64(current.End.Line), float64(r.End.Line)))
```

Comment:
`max(current.End.Line, r.End.Line)`

Reply:
Oh, thank goodness, what a wonderful addition to Go! :D